saleor/payment/utils.py

```diff
@@ -308,6 +308,8 @@ def gateway_postprocess(transaction, payment):
         payment.save(update_fields=changed_fields)
     transaction.already_processed = True
     transaction.save(update_fields=["already_processed"])
+    if order := payment.order:
+        order.update_total_paid()
```

Comment:
It adds an additional query for saving the captured_amount, event if we don't modify it. Why don't you check if `captured_amount in changed_fields`  and the call update_total_paid?